### PR TITLE
Adding PSR12 coding standard

### PR DESCRIPTION
**Pull request by ChrisB9**
Since PSR2 has been replaced with PSR12 and the source code is still fresh and little, i thought it might be good to provide this setup (Not much changes anyway).
Fits also good together with #52, since such a check could be automated in the future.

### CODE REVIEW

**Comment by tmotyl** [comment]
@ChrisB9 thanks for the PR, really appreciate!
How do you reformatted the files? Do you know if php_cs_fixer supports that, and what rules we need to add to the .php_cs.dist ?

File: Classes/Utility/ContentUtility.php [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?php
-declare(strict_types = 1);
+
+declare(strict_types=1);
```
Comment (tmotyl) [inline Classes/Utility/ContentUtility.php]: which rule from PSR-12 applies here? in TYPO3 Core it's formatted with spaces around "="

File: Classes/ContentObject/JsonContentObject.php [hunk 2]
```diff
@@ -167,7 +168,7 @@ protected function recursiveFind(array $haystack, $needle)
      * @param array $data
      * @return array
      */
-    protected function decodeFieldsIfRequired(array $data) : array
+    protected function decodeFieldsIfRequired(array $data): array
```
Comment (tmotyl) [inline Classes/ContentObject/JsonContentObject.php]: FYI, I've added a rule to check this one in https://github.com/TYPO3-Initiatives/headless/pull/58

**Comment by ChrisB9** [comment]
@tmotyl I did my checks with grumphp (pluswerk/grumphp-config does provide php_cs built-in with psr12 enabled). And yes i am aware that the php_cs does this and because you added it i wanted to already provide the base for it

File: Classes/Utility/ContentUtility.php (hunk 1 above)
Comment (ChrisB9) [reply Classes/Utility/ContentUtility.php]: My checks went entirely after PSR12, no additional special configuration
 

> Declare statements MUST contain no spaces and MUST be exactly declare(strict_types=1) (with an optional semi-colon terminator).
https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements

Comment (tmotyl) [reply Classes/Utility/ContentUtility.php]: you're right!

**Comment by ChrisB9** [comment]
Would it be okay, if i merge your #58 into my PR to incorporate all changes?
I'd like to add grumphp as well, if that aligns with your way of coding.

**Comment by tmotyl** [comment]
yes, you can merge or just copy paste the file content (I'll be squashing the commits before merging to master anyway).
The grumpphp will be a separate PR.

**Comment by lukaszuznanski** [comment]
Hi, first of all, thank you for contributing!

In #58 I think we have additional check for ```declare(strict_types = 1);``` so please get rid of it, as standard in PSR-12 is  ```declare(strict_types=1);```

**Comment by ChrisB9** [comment]
I have now updated the branch, but i realized, that i did get php-cs-fixer and phpcs mixed up. So you guys were totally right, the fixer does not yet support the new standard while phpcs does.
So if you want, we can hold of for these changes here for now, or keep them as references in... i am not yet sure how to proceed in that matter

File: .php_cs.dist [hunk 3]
```diff
@@ -23,39 +24,54 @@ if (PHP_SAPI !== 'cli') {
 return PhpCsFixer\Config::create()
     ->setRiskyAllowed(true)
     ->setRules([
+        '@DoctrineAnnotation' => true,
         '@PSR2' => true,
+        'array_syntax' => ['syntax' => 'short'],
+        'cast_spaces' => ['space' => 'none'],
+        'concat_space' => ['spacing' => 'one'],
+        'declare_equal_normalize' => ['space' => 'none'],
```
Comment (tmotyl) [inline .php_cs.dist]: make this
`'declare_equal_normalize' => ['space' => 'single'],`

so we've got PSR-12 compatible `declare(strict_types=1);` statements

File: Classes/ContentObject/JsonContentObject.php [hunk 4]
```diff
@@ -14,6 +11,10 @@
  *
  ***/
 
+declare(strict_types=1);
```
Comment (tmotyl) [inline Classes/ContentObject/JsonContentObject.php]: is this change intended (moving declare and namespace after comments?

Comment (tmotyl) [reply Classes/ContentObject/JsonContentObject.php]: ok, looks good then

**Comment by ChrisB9** [comment]
Yes, i just saw it and fixed that one directly

**Comment by tmotyl** [comment]
Thank you!

File: Classes/ContentObject/JsonContentObject.php (hunk 4 above)
Comment (GrahamCampbell) [reply Classes/ContentObject/JsonContentObject.php]: D'no how that rule got accepted to the standard. Can only assume the reviewers missed it. I've never seen any code before that followed that rule of having the doc above the declares. The 2 authors of the PSR must have wanted it, only...
